classes/machine-overrides-extender.bbclass

```diff
@@ -55,3 +55,22 @@ python machine_overrides_extender_handler() {
 
 machine_overrides_extender_handler[eventmask] = "bb.event.ConfigParsed"
 addhandler machine_overrides_extender_handler
+
+python machineoverrides_filtered_out_qa_handler() {
+    filtered_out = (d.getVar('MACHINEOVERRIDES_EXTENDER_FILTER_OUT') or "").split()
+    qa_error = d.getVar('MACHINEOVERRIDES_FILTERED_OUT_QA_ERROR')
+
+    for var in d.overridedata:
+        # We need to allow the overrides being used in the extender
+        # so avoid processing it.
+        if 'MACHINEOVERRIDES_EXTENDER' in var:
+            continue
+
+        for (r, o) in d.overridedata[var]:
+            common = list(set(o.split(":")).intersection(filtered_out))
+            if len(common) > 0:
+                raise bb.parse.SkipRecipe(qa_error % common)
```

Comment:
I'm not familiar with how this will appear to the user. Can you give an example of what the user will see if a variable has an override like `:mx8`?

Reply:
It looks like:
```console
$ bitbake glmark2
Loading cache: 100% |                                                         | ETA:  --:--:--
Loaded 0 entries from dependency cache.
Parsing recipes: 100% |########################################################| Time: 0:00:14
Parsing of 1916 .bb files complete (0 cached, 1916 parsed). 2900 targets, 413 skipped, 0 masked, 0 errors.
Removing 1 recipes from the imx8mq_evk sysroot: 100% |#########################| Time: 0:00:00
ERROR: Nothing PROVIDES 'glmark2'
glmark2 was skipped: ['mx8'] overrides cannot be used and need convertion to use the new BSP-specific overrides. Check 'meta-freescale/scripts/contrib/convert-bsp-specific-overrides'.

Summary: There was 1 ERROR message, returning a non-zero exit code.
```